### PR TITLE
[kealib] Cleanup, fix cmake config, [gdal] Add feature 'kea'

### PR DESCRIPTION
**Pull request by dg0yt**
Closes #31707.

### CODE REVIEW

File: ports/gdal/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "gdal",
-  "version-semver": "3.8.0",
+  "version-semver": "3.8.1",
+  "port-version": 1,
```
Comment (jimwang118) [inline ports/gdal/vcpkg.json:4]: It has been updated to a new version. Is this `port-version` keyword not needed?

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: ports/gdal/vcpkg.json (hunk 1 above)
Comment (dg0yt) [reply ports/gdal/vcpkg.json:4]: This PR is a draft, and it will remain in this state until #35428 is merged.

File: ports/kealib/portfile.cmake [hunk 2]
```diff
@@ -4,7 +4,9 @@ vcpkg_from_github(
     REF "kealib-${VERSION}"
     SHA512 82399f1332ff2aeb6342732e9e5c897c813109fd18e77cfc8d866f06adf4faa7f080f1f3c0a3b777fb3a679912dacf4851b7ad09a338d6087dd1d26eb2d1689f
     HEAD_REF master
-    PATCHES hdf5_include.patch
+    PATCHES
+        kealib-target.diff
```
Comment (dg0yt) [inline ports/kealib/portfile.cmake:8]: https://github.com/ubarsc/kealib/pull/45

**Comment by dg0yt** [comment]
paraview error:
~~~
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/lib/libpoppler.a(Parser.cc.o): in function `Parser::~Parser()':
Parser.cc:(.text+0x1c0): multiple definition of `Parser::~Parser()'; lib/libvisitcommon-pv5.11.a(Parser.C.o):/mnt/vcpkg-ci/buildtrees/paraview/src/a68035ab25-cb9320ae7a.clean/Utilities/VisItBridge/Library/VisItLib/common/parser/Parser.C:41: first defined here
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/lib/libpoppler.a(Parser.cc.o): in function `Parser::~Parser()':
Parser.cc:(.text+0x1c0): multiple definition of `Parser::~Parser()'; lib/libvisitcommon-pv5.11.a(Parser.C.o):/mnt/vcpkg-ci/buildtrees/paraview/src/a68035ab25-cb9320ae7a.clean/Utilities/VisItBridge/Library/VisItLib/common/parser/Parser.C:41: first defined here
~~~

Not really a regression, just never tested before `vcpkg-ci-gdal`:
Parser classes exist in poppler and in paraview.

**Comment by dg0yt** [comment]
> Parser classes exist in poppler and in paraview.

`Parser` in poppler isn't part of the public API, so it is possible to move the class to a unique namespace. There are more such classes: `Array`, `Dict`, ... but I don't want to touch them now.

**Review by jimwang118 — DISMISSED** [review]

**Comment by jimwang118** [comment]
When you're ready, please revert to "ready for review".

**Comment by jimwang118** [comment]
All features passed with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

**Pull request merged**